src/finally.js

```diff
@@ -103,9 +104,27 @@ Promise.prototype["finally"] = function (handler) {
                              finallyHandler);
 };
 
+
 Promise.prototype.tap = function (handler) {
     return this._passThrough(handler, TAP_TYPE, finallyHandler);
 };
 
+Promise.prototype.tapError = function (handlerOrPredicate) {
+    if(arguments.length === 1) {
+        return this._passThrough(handlerOrPredicate,
+                                 TAP_TYPE,
+                                 undefined,
+                                 finallyHandler);
+    } else {
+        var predicate = arguments[0];
```

Comment:
this needs to be typechecked  like .catch does https://github.com/petkaantonov/bluebird/blob/master/src/promise.js#L95-L99